mesonbuild/environment.py

```diff
@@ -902,6 +902,13 @@ def log_var(var, val):
         # this when the linker is stand-alone such as with MSVC C/C++, etc.
         link_flags = compile_flags + link_flags
 
+    if mesonlib.is_freebsd() and compiler.id == 'clang':
+        # If we're using Clang on FreeBSD also use lld (unless a linker has
+        # already been requested). lld is much more modern and capable than the
+        # default ld.bfd that is installed.
+        if not any(f.startswith('-fuse-ld') for f in link_flags):
+            link_flags += ['-fuse-ld=lld']
+
```

Comment:
I know ld.bfd on FreeBSD comes from a 10-year-old version of binutils, but I don't think it is good to change the default linker if users don't request it.

Reply:
Yes, and that 10 year old linker can't do things that users reasonably expect to do, if there is concensus to not use lld I'll drop this.

@jpakkane, @nirbheek: opinions?

Reply:
I think we should use the platform default tools unless they are fully broken. Changing default in secret is a simple way to get shouted at by distro developers. I think the obvious question here is why haven't they made that linker the default if the old one is so horribly broken (and also if they plan to make it the default any time soon)?

Reply:
It is impossible to pass a number of unit tests with the default linker, among others the version of bfd can't do lto. Should we error in that case or just let the build fail when it gets there?

Reply:
Which tests are failing with the default linker?

Reply:
I think this is the bug used to track the status of switching to lld on FreeBSD:
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=214864